### PR TITLE
Fixes for Go client unit test

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

There are several Go client unit test are not passing. Fixing them here.

### CODE REVIEW

**Comment by wolfstudy** [comment]
LGTM

File: pulsar-client-go/pulsar/consumer_test.go [hunk 1]
```diff
@@ -98,7 +100,10 @@ func TestConsumer(t *testing.T) {
 		assert.NotNil(t, msg)
 
 		assert.Equal(t, string(msg.Payload()), fmt.Sprintf("hello-%d", i))
-		assert.Equal(t, string(msg.Topic()), "persistent://public/default/my-topic")
+		assert.Equal(t, string(msg.Topic()), "persistent://public/default/" + topic)
```
Comment (wolfstudy) [inline pulsar-client-go/pulsar/consumer_test.go]:
```suggestion
		assert.Equal(t, msg.Topic(), "persistent://public/default/" + topic)
```

The type of msg.Topic itself is string, we don't need to convert here.

**Comment by merlimat** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**